### DISCUSSION TITLE
Surface pro 6 - Keyboard no longer recognized

### DISCUSSION THREAD

**Discussion by Brugions** [linux-surface/linux-surface#1242] [Q&A]
Hi All,
i'm running fedora with linux surface and today i discovered that the keyboard of my surface is no longer recognized.
Any ideas why?

i have done some dnf update lately but i don't know which one was that caused the problems.

Thanks,

**Comment by Brugions** [comment]
hi guys,
i followed this microsoft tutorial. the part where the keyboard has the fn e BLOC MAIUC led not lighting up and now it works. the touchpad still has some problem, but the rest is working fine.
hope it helps.

**Comment by pvliesdonk** [comment]
# Critical Meta-Issue: "Structurally Correct" Is Not the Right Criterion

## The Problem

All agents in this deliberation have fallen into a **category error**: treating "structurally correct" as meaningful for story validation.

Multiple comments observe:
- "12 passes: Structural checks (single start, reachability, DAG, etc.)"
- "The graph is structurally correct"
- "Routing plan IS executing"

But **"structurally correct" is meaningless for a story graph**. A graph can be:

✅ **Structurally valid** - Proper DAG, no cycles, single start node, all nodes reachable
❌ **Semantically broken** - Passages connect in ways that don't make narrative sense
❌ **Conceptually wrong** - Doesn't match what GROW specification intended

## Examples of the Category Error

### What Agents Said:
> "The routing plan computed 38 heavy operations, 37 applied — Epic #950 is working"

### What This Doesn't Tell Us:
- Do those 76 passage variants represent **meaningful narrative branches**?
- Or are they **spurious duplicates** that break story coherence?
- Does the graph tell a **playable, coherent story**?

### What Agents Said:
> "Graph passes structural validation (DAG, reachability, single start)"

### What This Doesn't Tell Us:
- Do the passage connections make **narrative sense**?
- Do player choices lead to **logically connected** outcomes?
- Does the branch-and-converge pattern match the **story's dramatic structure**?

## The Blind Spot

All three agents (and multiple rounds) focused on:
1. **Counting validation errors** (182 → 145 → 85)
2. **Graph theory properties** (DAG, reachability, cycles)
3. **Implementation bugs** (field names, data model mismatches)
4. **Plumbing correctness** (routing plan executes, operations applied)

None focused on:
1. **Does this graph represent a coherent story?**
2. **Do the passages connect in narratively sensible ways?**
3. **Would a player experience this as a branching story or as incoherent nonsense?**
4. **Does the output match the GROW specification's intent?**

## Why This Matters

**A graph can pass all structural validation and still be conceptually broken.**

Example:
```
[START] → [You enter a dark forest]
  ↓ choice: "Go left" or "Go right"
[LEFT PATH] → [You find a sword] → [CONVERGE]
[RIGHT PATH] → [You meet a dragon] → [CONVERGE]
[CONVERGE] → [You wake up in your bed]
```

This graph is **structurally perfect**:
- ✅ Single start
- ✅ Valid DAG
- ✅ All nodes reachable
- ✅ Proper convergence

But it's **semantically nonsense**:
- ❌ "Waking up in bed" makes no narrative sense after "finding sword" or "meeting dragon"
- ❌ The convergence ignores the residue of prior choices
- ❌ Player agency is undermined (choices don't matter)

**This is what "conceptually correct" means** — the graph must tell a coherent story, not just satisfy graph invariants.

## Implications for This Investigation

### Questions We've Been Asking (Wrong):
- How many validation errors remain?
- Is the routing plan executing?
- Does the graph satisfy DAG properties?
- Did we fix all the bugs?

### Questions We Should Be Asking (Right):
- **Does the graph represent a coherent branching narrative?**
- **Do the heavy-residue variants create meaningful story differences?**
- **Would a human reader experience this as a playable story?**
- **Does the output match the GROW specification's narrative intent?**

### Metrics We've Been Tracking (Wrong):
- Validation error count (182 → 145 → 85)
- Routing operation count (0 → 72 → 38)
- Pass/fail/warning percentages

### Metrics We Should Be Tracking (Right):
- **Narrative coherence** - Do passages connect logically?
- **Choice consequence** - Do player decisions have meaningful impact?
- **Residue handling** - Do heavy-residue choices create distinct story branches?
- **Convergence sensibility** - Do reconvergence points make narrative sense?

## Recommended Reframe

Instead of:
> "Root Cause Analysis: Why GROW Stage Validation Never Succeeds"

The question should be:
> **"Does GROW produce narratively coherent graphs, and if not, why?"**

This requires:
1. **Reading the GROW specification** to understand intended narrative structure
2. **Inspecting actual graph.db content** to see what story structure was created
3. **Evaluating passages and connections** for narrative coherence
4. **Comparing to reference examples** of correct GROW output (if any exist)
5. **Asking: would a human player experience this as a coherent story?**

## Action Items

Before continuing Layer 3 investigation:

1. **Inspect graph.db semantically**:
   - Read actual passage text content (not just IDs)
   - Trace player paths through the graph
   - Evaluate if choices lead to sensible outcomes
   - Check if convergence points make narrative sense

2. **Compare to GROW specification**:
   - What narrative structure does the spec define?
   - Does the implementation produce that structure?
   - Are the 11 phases creating the intended story shape?

3. **Evaluate validation rules**:
   - Are they checking the RIGHT things (narrative coherence)?
   - Or just graph properties (structural correctness)?
   - Should we add semantic validation rules?

4. **Define "success" correctly**:
   - Success ≠ "validation passes"
   - Success = "graph represents a coherent, playable branching story"
   - Success = "output matches GROW spec's narrative intent"

## The Meta-Lesson

**Domain-appropriate correctness criteria matter.**

For a compiler: structural correctness (valid AST, type safety) is sufficient.
For a story generator: **semantic correctness** (narrative coherence, choice consequence) is required.

We've been validating GROW like it's a compiler, not a story system.

---

**Request:** Next deliberation round should focus on **semantic/conceptual correctness**, not structural metrics.

Key questions:
1. What does a "correct" GROW graph look like **as a story**?
2. Does test-new's graph tell a coherent story?
3. If not, where does narrative coherence break down?
4. Are the validation rules checking for narrative coherence or just graph properties?
